octopus-server/src/main/java/octopus/teamcity/server/generic/GenericParameterProcessor.java

```diff
@@ -15,36 +15,51 @@
 
 package octopus.teamcity.server.generic;
 
-import java.util.Collection;
-import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
+import com.google.common.collect.Lists;
+import com.intellij.openapi.util.text.StringUtil;
 import jetbrains.buildServer.serverSide.InvalidProperty;
 import jetbrains.buildServer.serverSide.PropertiesProcessor;
 import octopus.teamcity.common.commonstep.CommonStepPropertyNames;
 
 public class GenericParameterProcessor implements PropertiesProcessor {
 
+  private static final CommonStepPropertyNames KEYS = new CommonStepPropertyNames();
+
   @Override
-  public Collection<InvalidProperty> process(final Map<String, String> properties) {
+  public List<InvalidProperty> process(final Map<String, String> properties) {
+    final List<InvalidProperty> failedProperties = Lists.newArrayList();
+
     final String stepType = properties.get(CommonStepPropertyNames.STEP_TYPE);
 
     if (stepType == null) {
-      return Collections.singletonList(
+      failedProperties.add(
           new InvalidProperty(CommonStepPropertyNames.STEP_TYPE, "No StepType specified"));
```

Comment:
nit: Commonality of error messaging "StepType vs steptype" (vs step type) see line 50